src/goto-programs/remove_returns.cpp

```diff
@@ -46,74 +51,91 @@ class remove_returnst
   void undo_function_calls(
     goto_functionst &goto_functions,
     goto_programt &goto_program);
+
+  symbol_exprt get_or_create_return_value_symbol(const irep_idt &function_id);
```

Comment:
For future reference: it would have been nice to place the refactoring (which is where this function came about) in a separate commit. I believe this refactoring is a big improvement, but also it caused the biggest code churn here.